open-metadata-implementation/access-services/subject-area/subject-area-api/src/main/java/org/odpi/openmetadata/accessservices/subjectarea/properties/objects/graph/Line.java

```diff
@@ -92,35 +85,44 @@ public Line(String name, LineType lineType) {
-
-        RelationshipDef relationshipDef =archiveAccessor.getRelationshipDefByGuid(this.typeDefGuid);
-        RelationshipEndDef endDef1 = relationshipDef.getEndDef1();
-        RelationshipEndDef endDef2 = relationshipDef.getEndDef2();
-
-        String end1TypeName = endDef1.getEntityType().getName();
-        String end2TypeName = endDef2.getEntityType().getName();
+//        OpenMetadataTypesArchiveAccessor archiveAccessor = OpenMetadataTypesArchiveAccessor.getInstance();
+//
+//        RelationshipDef relationshipDef =archiveAccessor.getRelationshipDefByGuid(this.typeDefGuid);
+//        RelationshipEndDef endDef1 = relationshipDef.getEndDef1();
+//        RelationshipEndDef endDef2 = relationshipDef.getEndDef2();
+//
+//        String end1TypeName = endDef1.getEntityType().getName();
+//        String end2TypeName = endDef2.getEntityType().getName();
         // Maybe consider subtypes here? We cannot use the Repositoryhelper isTypeOf because Line is used client side
         // which does not have access to the repository helper.
-        if (end1TypeName.equals("GlossaryTerm")) {
-            end1TypeName = "Term";
-        }
-        if (end2TypeName.equals("GlossaryTerm")) {
-            end2TypeName = "Term";
-        }
+//        if (end1TypeName.equals("GlossaryTerm")) {
+//            end1TypeName = "Term";
+//        }
+//        if (end2TypeName.equals("GlossaryTerm")) {
+//            end2TypeName = "Term";
+//        }
+//        getLineEnd1();
+//        getLineEnd1();
+//
+//        LineEnd lineEnd1 = new LineEnd(getEnd1TypeName(),
+//                                       getEnd1AttributeName(),
+//                                       getEnd1AttributeDescription(),
+//                                       endDef1.getAttributeCardinality()
+//                                       );
+//        LineEnd lineEnd2 = new LineEnd(end2TypeName,
+//                                       endDef2.getAttributeName(),
+//                                       endDef2.getAttributeDescription(),
+//                                       endDef2.getAttributeCardinality()
+//                                       );
+        setEnd1(getLineEnd1());
+        setEnd2(getLineEnd2());
+    }
 
-        LineEnd lineEnd1 = new LineEnd(end1TypeName,
-                                       endDef1.getAttributeName(),
-                                       endDef1.getAttributeDescription(),
-                                       endDef1.getAttributeCardinality()
-                                       );
-        LineEnd lineEnd2 = new LineEnd(end2TypeName,
-                                       endDef2.getAttributeName(),
-                                       endDef2.getAttributeDescription(),
-                                       endDef2.getAttributeCardinality()
-                                       );
-        setEnd1(lineEnd1);
-        setEnd2(lineEnd2);
+    protected LineEnd getLineEnd1() {
+       return null;
```

Comment:
May be make Line abstract and make these methods are abstract?

Reply:
> May be make Line as abstract and make these methods abstract?

I had a play with this idea, the OmrsLine extends from Line and would need implementations of these methods that does not make sense for it. I have left this as is - as I do not want to change this without thinking it through in more detail what is required for OMRSLine.

Reply:
When do you want to think about it? I'm afraid this will remain so and in the future you can bring us trouble.

Reply:
> > May be make Line as abstract and make these methods abstract?
> 
> I had a play with this idea, the OmrsLine extends from Line and would need implementations of these methods that does not make sense for it. I have left this as is - as I do not want to change this without thinking it through in more detail what is required for OMRSLine.

When do you want to think about it? I'm afraid this will remain so and in the future you can bring us trouble.

Reply:
> OMRSLine

It turned out to be straight forward, this OMRSLine was not used in the code so I have deleted and made the methods abstract. Thanks @immmus